SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java

```diff
@@ -142,21 +158,42 @@ public MailTemplateHelpers(SandboxManager sandboxManager, SingularityConfigurati
     }
   }
 
-  private Optional<Long> getMaybeOffset(String slaveHostname, String fullPath, String filename, Optional<String> directory) {
+  // Searches through the file,
+  private Optional<Long> getMaybeOffset(final String slaveHostname, final String fullPath, final Long logLength, Optional<SingularityTask> task) {
+    long offset = 0;
+    String regex = getLogErrorRegex(task);
+    long length = logLength + regex.length(); // Get extra so that we can be sure to find the error
```

Comment:
If we fetch just the amount we're supposed to, then it's possible that the match will be half in one fetched chunk and half in the next, in which case we wouldn't match it.
For example, we're looking for 'ERROR'. If we fetch 10 bytes, and the log file is 'xxxxxxxxERRORxxxxxxx', we would fetch 'xxxxxxxxER', not match, then fetch 'RORxxxxxxx', and still not match. Whereas with the extra length we'd fetch 'xxxxxxxxERRORxx' and correctly find the match.